modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.jsx

```diff
@@ -310,11 +282,70 @@ const SetupAnalyticsCloudPage = ({
 					);
 				}
 			}
+		};
+
+		try {
+			setIsLoadingSubmitButton(true);
+
+			if (featureFlags.includes('LPS-159127')) {
+				await actRaysourceContact(
+					removeContactRoleRaysource,
+					removeHighPriorityContact,
+					project,
+					sessionId,
+					provisioningServerAPI
+				);
+				await actRaysourceContact(
+					associateContactRoleRaysource,
+					addHighPriorityContact,
+					project,
+					sessionId,
+					provisioningServerAPI
+				);
+				await actLiferayContact(
+					addHighPriorityContact,
+					associateContactRoleLiferay,
+					project,
+					client
+				);
+				await actLiferayContact(
+					removeHighPriorityContact,
+					removeContactRoleLiferay,
+					project,
+					client
+				);
+			}
+
+			handleDataSubmit();
 			setIsLoadingSubmitButton(false);
 
 			handlePage(true);
-		} catch {
-			setIsLoadingSubmitButton(false);
+		} catch (error) {
```

Comment:
@pedroliveirav if we are properly comparing old and new contact arrays wouldnt we exclude the possibility of hitting the conflict code? or what circumstance would allow us to see this error in the ui?

Separate note, we should either way write playwrite tests to test the conflict edge case and ensure we either dont get the error or it shows up when expected

Reply:
All the contacts handled are on the portal side, but the conflict occurs when the user assigned to a specific role already has that role in the provisioning, so when the error is a conflict, the requests made are only on the liferay side. I don't know if my answer satisfies all your doubts.